FantasyUniverse/Scenes/ResetPassword/ResetPasswordViewModel.swift

```diff
@@ -13,23 +13,22 @@ class ResetPasswordViewModel: ObservableObject {
     private let validator = AuthenticationValidator()
     private let firebaseAuthService = FirebaseAuthService()
     @Published var emailFieldValue = ""
-    @Published var errorMessage = ""
+    @Published var validationErrorMessage = ""
+    @Published var fieldsValid = false
```

Comment:
Again, having a closed set of mutually exclusive states would be much better. So what something like an enum with a case for states like `initial`, `error`, `valid` and appropriate associated values would do is, it would ensure multiple of these values cannot be active at the same time and the combinations of active values are easier to reason about. For example, what does a `fieldsValid` == false, `validationErrowMessage` == "something", `isResetSuccesful` == true mean?